src/main/java/org/elasticsearch/index/query/IdsQueryParser.java

```diff
@@ -107,26 +98,13 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                 }
             }
         }
-
-        if (!idsProvided) {
-            throw new QueryParsingException(parseContext, "[ids] query, no ids values provided");
-        }
-
         if (ids.isEmpty()) {
-            return Queries.newMatchNoDocsQuery();
-        }
-
-        if (types == null || types.isEmpty()) {
-            types = parseContext.queryTypes();
-        } else if (types.size() == 1 && Iterables.getFirst(types, null).equals("_all")) {
-            types = parseContext.mapperService().types();
-        }
-
-        TermsQuery query = new TermsQuery(UidFieldMapper.NAME, Uid.createTypeUids(types, ids));
-        query.setBoost(boost);
-        if (queryName != null) {
-            parseContext.addNamedQuery(queryName, query);
+            throw new QueryParsingException(parseContext, "[ids] query, no ids values provided");
```

Comment:
Talked to clint about this, the difference here is empty ids list, which is okay and should return no docs, and missing "value" field in query dsl, which throws an exception, but this is only affecting parsing the json. So I would opt for continue throwing the parsing exception, but validate() which operated on the query object should allow empty ids list.